src/openforms/registrations/api/serializers.py

```diff
@@ -38,21 +38,14 @@ class RegistrationAttributeSerializer(serializers.Serializer):
     )
 
 
-class InformatieObjectTypeSerializer(APIModelSerializer):
-    class Meta:
-        model = InformatieObjectType
-        fields = (
-            "url",
-            "omschrijving",
-        )
-
-
-class CatalogusDomainSerializer(APIModelSerializer):
+class CatalogusSerializer(APIModelSerializer):
     class Meta:
         model = Catalogus
-        fields = ("domein",)
+        fields = ("url", "domein", "rsin")
 
 
-class InformatieObjectTypeChoiceSerializer(serializers.Serializer):
-    informatieobjecttype = InformatieObjectTypeSerializer()
-    catalogus = CatalogusDomainSerializer()
+class InformatieObjectTypeSerializer(serializers.Serializer):
+    catalogus_domein = serializers.CharField(label=_("catalogus domein"))
+    catalogus_rsin = serializers.CharField(label=_("catalogus rsin"))
+    url = serializers.URLField(label=_("url"))
+    omschrijving = serializers.CharField(label=_("omschrijving"))
```

Comment:
discussed in meeting - probably best to move this to `openforms.contrib.zgw`, and subclass the `Catalogus` model to add an extra field for `naam`